modules/cnf-provisioning-real-time-and-low-latency-workloads.adoc

```diff
@@ -278,4 +283,85 @@ For more information, see link:https://access.redhat.com/documentation/en-us/ope
 [id="performance-addon-operator-scheduling-workload-onto-worker-with-real-time-capabilities_{context}"]
 == Scheduling a workload onto a worker with real-time capabilities
 
-Use label selectors that match the nodes attached to the machine config pool that was configured for low latency by the Performance Addon Operator. For more information, see link:https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/[Assigning pods to nodes].
+Use label selectors that match the nodes attached to the machine config pool that was configured for low latency by the Performance Addon controller. For more information, see link:https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/[Assigning pods to nodes].
+
+[id="node-tuning-operator-disabling-CPUs-for-power-consumption_{context}"]
+== Reducing power consumption by taking CPUs offline
+
+You can generally anticipate telecommunication workloads. When not all of the CPU resources are required, the Performance Addon conroller allows you take unused CPUs offline to reduce power consumption by manually updating the performance profile.
+
+To take unusued CPUs offline, you must perform the following tasks:
+
+. Update the performance profile.
+
+.. Review the YAML file by running the following command:
++
+[source,terminal]
+----
+$ cat my-performance-profile.yaml
+----
++
+.Example output
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+  name: performance
+spec:
+  additionalKernelArgs:
+  - nmi_watchdog=0
+  - audit=0
+  - mce=off
+  - processor.max_cstate=1
+  - intel_idle.max_cstate=0
+  - idle=poll
+  cpu:
+    isolated: "2-23,26-47"
+    reserved: "0,1,24,25,48-59"
+    offlined: "" <1>
+  nodeSelector:
+    node-role.kubernetes.io/worker-cnf: ""
+  numa:
+    topologyPolicy: single-numa-node
+  realTimeKernel:
+    enabled: true
+----
+<1> Optional. You can list CPUs in the `offlined` field to take the specified CPUs offline.
+
+.. Set the offline CPUs in the performance profile and save the contents of the file.
++
+.Example updated performance profile
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+  name: performance
+spec:
+  additionalKernelArgs:
+  - nmi_watchdog=0
+  - audit=0
+  - mce=off
+  - processor.max_cstate=1
+  - intel_idle.max_cstate=0
+  - idle=poll
+  cpu:
+    isolated: "2-23,26-47"
+    reserved: "0,1,24,25"
+    offlined: “48-59” <1>
+  nodeSelector:
+    node-role.kubernetes.io/worker-cnf: ""
+  numa:
+    topologyPolicy: single-numa-node
+  realTimeKernel:
+    enabled: true
+----
+<1> Lists the CPUs taken offline.
```

Comment:
Maybe this is my ignorance. The CPUs listed here will be taken off line as soon as you `oc apply`? Or the NTO is allowed to take these CPUs offline if they are not needed?

Reply:
The change will update the generated Machine Config and that will cause a reboot immediately unless the MC is paused.

Reply:
@MarSik @amolnar-rh  I apologize for not being more clear. It is not clear to me what the procedure does. 
Is the user taking the specified CPUs off line by editing the MC? 
Or, is the user configuring things so that the NTO can take those CPUs off line as it wants?

Reply:
@MarSik @marioferh Could you please take a look and clarify?

Reply:
the flow is, 
- With PerformanceProfileCreator you can generate a Performance Profile (or by hand, but much better with creator)
- label the node
- unpause mcp in order to apply changes
- mcp pics mc
- mc is applied and some system reboots can be done
- 
https://docs.openshift.com/container-platform/4.6/scalability_and_performance/cnf-performance-addon-operator-for-low-latency-nodes.html#performance-addon-operator-provisioning-worker-with-real-time-capabilities_cnf-master